llvm/test/lab3/simonyan/main_test.mir

```diff
@@ -0,0 +1,347 @@
+  
+  for.body3:                                        ; preds = %for.cond1
+    %4 = load i32, ptr %i, align 4
+    %5 = load i32, ptr %res, align 4
+    %add = add nsw i32 %5, %4
+    store i32 %add, ptr %res, align 4
+    %6 = load i32, ptr %j, align 4
+    %7 = load i32, ptr %res, align 4
+    %add4 = add nsw i32 %7, %6
+    store i32 %add4, ptr %res, align 4
+    br label %for.inc
+  
+  for.inc:                                          ; preds = %for.body3
+    %8 = load i32, ptr %j, align 4
+    %inc = add nsw i32 %8, 1
+    store i32 %inc, ptr %j, align 4
+    br label %for.cond1, !llvm.loop !7
+  
+  for.end:                                          ; preds = %for.cond1
+    br label %for.inc5
+  
+  for.inc5:                                         ; preds = %for.end
+    %9 = load i32, ptr %i, align 4
+    %inc6 = add nsw i32 %9, 1
+    store i32 %inc6, ptr %i, align 4
+    br label %for.cond, !llvm.loop !8
+  
+  for.end7:                                         ; preds = %for.cond
+    %10 = load i32, ptr %res, align 4
+    ret i32 %10
+  }
+  
+  attributes #0 = { mustprogress noinline nounwind optnone uwtable "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
+  
+  !llvm.module.flags = !{!0, !1, !2, !3}
+  !llvm.ident = !{!4}
+  
+  !0 = !{i32 1, !"wchar_size", i32 2}
+  !1 = !{i32 8, !"PIC Level", i32 2}
+  !2 = !{i32 7, !"uwtable", i32 2}
+  !3 = !{i32 1, !"MaxTLSAlign", i32 65536}
+  !4 = !{!"clang version 17.0.6 (https://github.com/SSuren4ik/llvm-nnsu-2024.git 7d2f0dadb28ce8c42178a65b334c1d968eeb9ffa)"}
+  !5 = distinct !{!5, !6}
+  !6 = !{!"llvm.loop.mustprogress"}
+  !7 = distinct !{!7, !6}
+  !8 = distinct !{!8, !6}
+
+...
+---
+name:            _Z10empty_funcv
+alignment:       16
+tracksRegLiveness: true
+tracksDebugUserValues: true
+frameInfo:
+  maxAlignment:    8
+  maxCallFrameSize: 0
+machineFunctionInfo: {}
+body:             |
+  bb.0.entry:
+    ; CHECK: MOV64mr %0, &ic
+    ; CHECK: %0:gr64 = ADD64ri32 %0, 1, implicit-def $eflags
+    RET64
+
+...
+---
+name:            _Z9main_funciii
+alignment:       16
+tracksRegLiveness: true
+hasWinCFI:       true
+tracksDebugUserValues: true
+liveins:
+  - { reg: '$ecx' }
+  - { reg: '$edx' }
+  - { reg: '$r8d' }
+frameInfo:
+  stackSize:       24
+  offsetAdjustment: -24
+  maxAlignment:    8
+  maxCallFrameSize: 0
+stack:
+  - { id: 0, name: a.addr, offset: -12, size: 4, alignment: 4 }
+  - { id: 1, name: b.addr, offset: -16, size: 4, alignment: 4 }
+  - { id: 2, name: c.addr, offset: -20, size: 4, alignment: 4 }
+  - { id: 3, name: d, offset: -28, size: 4, alignment: 4 }
+  - { id: 4, name: i, offset: -24, size: 4, alignment: 4 }
+machineFunctionInfo: {}
+body:             |
+  bb.0.entry:
+    liveins: $ecx, $edx, $r8d
+  
+    $rsp = frame-setup SUB64ri32 $rsp, 24, implicit-def dead $eflags
```

Comment:
What command do you use for generating input MIR? Looks like your MIR already contains physical registers while your pass adds virtual ones. It requires either to [re-]allocate registers additionally or make changes before register allocation pass.